.github/workflows/release.yml

```diff
@@ -1,16 +1,46 @@
-name: release
+name: 'New Release'
+run-name: 'Release ${{ inputs.version_number }}'
+
 on:
-  push:
-    tags:
-      - 'v*'
-jobs: 
-  goreleaser:
+  workflow_dispatch:
+    inputs:
+      version_number:
+        description: 'Version number (e.g., v1.0.0, v1.0.0-pre, v1.0.0-pre1)'
+        required: true
+      skip_tests:
+        description: 'Skip QA acceptance tests, define value to `true` to explicitly skip'
+
+jobs:
+
+  validate-version-input:
     runs-on: ubuntu-latest
+    steps:
+      - name: Validation of version format
+        run: |
+          echo "${{ inputs.version_number }}" | grep -P '^v\d+\.\d+\.\d+(-pre[A-Za-z0-9-]*)?$'  
+
+    # QA acceptance tests are skipped when explicit input parameter is used
+  run-qa-acceptance-tests:
+    needs: [ validate-version-input ]
+    if: needs.validate-version-input.result == 'success' && inputs.skip_tests != 'true'
+    secrets: inherit
+    uses: ./.github/workflows/acceptance-tests.yml
+    with:
+      atlas_cloud_env: "qa"
+
+  release:
+    runs-on: ubuntu-latest
+    needs: [ validate-version-input, run-qa-acceptance-tests ]
+    if: ${{ always() && needs.validate-version-input.result == 'success' && (needs.run-qa-acceptance-tests.result == 'skipped' || needs.run-qa-acceptance-tests.result == 'success') }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
       - name: Unshallow
         run: git fetch --prune --unshallow
+      - name: Create release tag
+        uses: rickstaa/action-create-tag@v1.7.2 # will fail if existing tag is present
```

Comment:
I've found this action https://github.com/mathieudutour/github-tag-action (520 starts) which seems to be used more than rickstaa/action-create-tag (60 starts). Maybe we could use that one (not a strong opinion)

Reply:
Yes that was an option I looked into, went for https://github.com/rickstaa/action-create-tag because it had the feature of signing tags (even though we are not using it yet) which this other action did not have.

Reply:
it makes sense. Thanks!